### ISSUE TITLE
Processing issues with OpenMC0.14 / NJOY2016

### ISSUE THREAD

**Issue by shimwell** [IAEA-NDS/FENDL-ENDF#1]
Thanks for making this repo I had a look around and really like the setup. 

Processing FENDL files with openmc and the openmc interface to NJOY raises the following errors:

This could be bugs on the openmc or NJOY side so I'm keen to spend some more time looking into them.

- [x] openmc_data/fendl-3.2a-endf/neutron/endf/n_0131_1-H-3.endf 'utf-8' codec can't decode byte 0x92 in position 7137: invalid start byte
- [x] openmc_data/fendl-3.2a-endf/neutron/endf/n_0825_8-O-16.endf invalid literal for int() with base 10: '0         0'
- [x] openmc_data/fendl-3.2a-endf/neutron/endf/n_4843_48-Cd-112.endf invalid literal for int() with base 10: ''
- [ ] openmc_data/fendl-3.2a-endf/neutron/endf/n_9228_92-U-235.endf Command 'njoy' returned non-zero exit status 2.
- [ ] openmc_data/fendl-3.2-endf/neutron/endf/n_9228_92-U-235.endf Command 'njoy' returned non-zero exit status 2.
- [x] openmc_data/fendl-3.2-endf/neutron/endf/n_4843_48-Cd-112.endf invalid literal for int() with base 10: ''
- [ ] openmc_data/fendl-3.1d-endf/n_0628_6-C-13.txt Command 'njoy' returned non-zero exit status 77.
- [ ] openmc_data/fendl-3.1d-endf/n_0831_8-O-18.txt Command 'njoy' returned non-zero exit status 77.
- [ ] openmc_data/fendl-3.1d-endf/n_9228_92-U-235.txt Command 'njoy' returned non-zero exit status 2.
- [ ] openmc_data/fendl-3.1a-endf/n_9228_92-U-235.txt Command 'njoy' returned non-zero exit status 2.
- [ ] openmc_data/fendl-3.1a-endf/n_0831_8-O-18.txt Command 'njoy' returned non-zero exit status 77.
- [ ] openmc_data/fendl-3.1a-endf/n_0628_6-C-13.txt Command 'njoy' returned non-zero exit status 77.
- [ ] openmc_data/fendl-3.0-endf/endf/n_0628_6-C-13.txt Command 'njoy' returned non-zero exit status 77
- [x] openmc_data/fendl-3.2-endf/neutron/endf/n_0131_1-H-3.endf 'utf-8' codec can't decode byte 0x92 in position 7137: invalid start byte
- [x] openmc_data/fendl-3.1d-endf/n_0131_1-H-3.txt 'utf-8' codec can't decode byte 0x92 in position 7137: invalid start byte
- [ ] openmc_data/fendl-3.1a-endf/n_0131_1-H-3.txt 'utf-8' codec can't decode byte 0x92 in position 7137: invalid start byte
- [x] openmc_data/fendl-3.0-endf/endf/n_0131_1-H-3.txt 'utf-8' codec can't decode byte 0x92 in position 7137: invalid start byte

This can be reproduced by running these commands after installing the openmc_data package openmc and njoy
```
conda install -c conda-forge openmc njoy2016
pip install openmc_data
generate_fendl -r 3.2a
generate_fendl -r 3.2
generate_fendl -r 3.1d
generate_fendl -r 3.1a
generate_fendl -r 3.0
generate_fendl -r 3.2b
```

essentially these scripts download the fendl data then call ```openmc.data.IncidentNeutron().from_endf(filename)``` with all the endf text files and some of the files cause errors

On a slightly related topic I would be keen to add some CI to this repo, perhaps it could run some of these processing tools to check things are working on each change

**Comment by gschnabel** [comment]
Hi Jon

Thank you for reporting these issues.

Some of the reported files files had indeed formatting issues. These files are:

- n_0131_1-H-3.endf
- n_0825_8-O-16.endf
- n_4843_48-Cd-112.endf

I was able to fix their issues. I don't want to rewrite history so I guess it is best to create extra branches that branch off from the respective FENDL version. For instance, a branch "FENDL-3.2a-fixed" that branches off from FENDL-3.2a. What I am asking myself at present is whether I should move the original tag, e.g., FENDL-3.2a, to the tip of the branch with the fix. I don't see any reason not to do it because what is the point of keeping a reference to two versions where the only difference between them is that one version contains small bugs in some files that prevent their proper processing into application forms. If you have a different perspective, let me know your considerations.

Regarding the issues listed for FENDL-3.1a, the situation with FENDL-3.1 subversions was quite messy (even messier than the overall version naming scheme; we plan to improve that in the future) and, to my knowledge, we neither have FENDL-3.1a in the GitHub repo nor on nds.iaea.org/fendl website anymore. If you are still able to download it from some IAEA location, let me know. If you have archived it somewhere else,  maybe it is better to discard that version from the openmc.data repository as well to avoid confusion.

I couldn't find any formatting issues in the other files n_0628_6-C-13.txt, n_0831_8-O-18.txt, n_9228_92-U-235.txt. I will try in the next days to process them with the processing pipeline bundled in the FENDL-processed pipeline (code/process_fendl.sif) and see if I encounter any problems. The NJOY2016 version we are using is tweaked compared to the official one to solve some processing issues, so maybe this is the case here. I will get back to this discussion thread once I know more. 

Adding CI to this repo is a great idea. Shall we open another discussion thread here or do you prefer discussing the options via video chat?

**Comment by gschnabel** [comment]
I think that all issues reported have been resolved in FENDL-3.2b and FENDL-3.2c. Feel free to reopen if an issue hasn't been resolved.
